### DISCUSSION TITLE
how can I build a new mysql image ?

### DISCUSSION THREAD

**Discussion by linayang999** [kubeflow/pipelines#10862] [General]
as the latest mysql version kubeflow pipeline using is  gcr.io/ml-pipeline/mysql:8.0.26 which is based on outdated debian 10 os version , and this image is not allowed to use as it's out of support for a time .

may I know if I want to build a new pipeline mysql image , any dockerfile can refer to  or any mysql image which is built based on debian 12  I can use now ？ what I want is a new iamge built on debian 12

thanks for any support ,thanks .

the mysql manifest:
https://github.com/kubeflow/pipelines/blob/45f3457d4206d1793ebc4ab60f188fe2a17db4eb/manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml#L51

**Comment by rimolive** [comment]
Did you try out the official MySQL image? https://hub.docker.com/_/mysql

You can also try replace with a MariaDB image: https://hub.docker.com/_/mariadb

**Reply by linayang999** [reply]
thanks for your quick respose  @rimolive 
acturally , the current manifest we are using is kubeflow1.5 ,which with the pipeline mysql vsersion is mysql5.7  https://github.com/kubeflow/pipelines/blob/6cddf73a1e79540aeacb761e517fd4319b3b3aa5/manifests/kustomize/third-party/mysql/base/mysql-deployment.yaml#L29C1-L29C44
I was just wondering if it's compatible if I used the latest 8.0* debian mysql image(mysql:8.0.37-debian) to adopt kubeflow 1.5's manifest ?  or any change I need to update?  and I checked the image for version 5.7 from gcr is different with the  offical mysql image , was there any customermization for gcr image which we are currently using ?
![ecdd659780cb717efa6f139b5bed1ab](https://github.com/kubeflow/pipelines/assets/96032364/55f2b4a5-3e25-4323-b960-9cb40d7aba96)

**Reply by rimolive** [reply]
Ok, I get it. You are free to upgrade the MySQL version, but beginning KFPv2 you must use MySQL 8.0 anyway, and the reason is MySQL does not support window functions, which is used in some KFPv2 queries.

The way MySQL image from KFP is built is different in terms of env vars to be used to configure root password, database name, etc. That said, if using official MySQL image, expect to make changes in the manifests to use the corresponding env var from the official image.
